base/dates/arithmetic.jl

```diff
@@ -86,11 +86,11 @@ for op in (:+, :-)
     @eval begin
         ($op){T<:TimeType}(x::AbstractArray{T}, y::GeneralPeriod) = broadcast($op, x, y)
         ($op){P<:GeneralPeriod}(y::TimeType, x::StridedArray{P}) = broadcast($op, x, y)
```

Comment:
the argument order is y, x on the left and x, y on the right so it only works for commutative ops